api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -61,7 +65,7 @@
  * <p>This needs to implement an interface to support Transactional
  */
 @Service
-public class WorkspaceServiceImpl implements WorkspaceService {
+public class WorkspaceServiceImpl implements WorkspaceService, GaugeDataCollector {
```

Comment:
In writing up a test, I'm starting to think this approach is [asking for trouble](https://stackoverflow.com/questions/39498827/spring-service-implementing-multiple-interfaces-good-or-bad). Basically, while Spring is happy to inject the (same instance of the) service as either bean, it's tricky to mock them separately. One way to do it is to add qualifiers, which is probably the safest and sanest thing to do if we want to keep multiple interfaces on the same service.

Adding the `GaugeDataCollector` as a parent interface to `WorkspaceService` would work, but is a nasty hack and doesn't reflect the design intent we have here.

Another option would be to have a separate implementation of `GaugeDataCollector` for each gauge collector, and then distinguish those by qualifier. Pretty bloated, but it fits our patterns better. Those could then depend on the services (or data managers, if we had those) to hit the daos.

Reply:
I'm pretty satisfied for now. I can handle any ambiguity by adding qualifiers if we care.